src/networking.c

```diff
@@ -3752,3 +3863,52 @@ int handleClientsWithPendingReadsUsingThreads(void) {
 
     return processed;
 }
+
+/* Returns true if client memory limit was reached */
+int clientEvictionCheckLimit() {
+    size_t maxmemory_clients_actual = SIZE_MAX;
+
+    /* Handle percentage of maxmemory*/
+    if (server.maxmemory_clients < 0 && server.maxmemory > 0)
+        maxmemory_clients_actual = (size_t)((double)server.maxmemory * -(double)server.maxmemory_clients / 100);
```

Comment:
Truncation may occur in 32 bits, `maxmemory` is `ULLONG_MAX`.

Reply:
I see. But why is `maxmemory` an `unsigned long long`? Shouldn't it be a `size_t`. How can I have more than 4g mem on 32bit?

Reply:
Yes, I don't understand why `maxmemory` is `unsigned long long` either.

Reply:
i don't know.. maybe historical reasons, but i think it's safer to keep it this way.

Reply:
Added range check. I think we should change to `size_t` in the near future as part of 7.0.